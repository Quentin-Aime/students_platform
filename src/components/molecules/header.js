import React from "react";
import styled from "styled-components";
import { Thumbnail } from "../atoms/thumbnail";
import group from "../../assets/images/ramen.svg";
import aymeline from "../../assets/images/aymeline.svg";
import melanie from "../../assets/images/melanie.svg";
import marie from "../../assets/images/marie.svg";
import quentin from "../../assets/images/quentin.svg";
import rainald from "../../assets/images/rainald.svg";
import federica from "../../assets/images/federica.svg";
import mathieu from "../../assets/images/mathieu.svg";
import ketsia from "../../assets/images/ketsia.svg";

export const Header = ({ setMember }) => {
  const members = [
    {
      id: 1,
      imgUrl: group,
    },
    { id: 2, imgUrl: melanie },
    { id: 3, imgUrl: marie },
    { id: 4, imgUrl: aymeline },
    { id: 5, imgUrl: quentin },
    { id: 6, imgUrl: rainald },
    { id: 7, imgUrl: federica },
    { id: 8, imgUrl: mathieu },
    { id: 9, imgUrl: ketsia },
  ];
  return (
    <HeaderContainer>
      {members.map((member) => (
        <Thumbnail
          imgUrl={member.imgUrl}
          id={member.id}
          setMember={setMember}
        />
      ))}
    </HeaderContainer>
  );
};

const HeaderContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
`;
