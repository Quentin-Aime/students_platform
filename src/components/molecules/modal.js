import React from "react";
import Modal from "react-modal";
import { Title } from "../atoms/title";
import { Picture } from "../atoms/picture";
import { Text } from "../atoms/text";

export const CustomModal = ({
  title,
  thumbnail,
  description,
  detailsTitle,
  details,
  isOpen,
  onRequestClose,
}) => {
  return (
    <Modal isOpen={isOpen} onRequestClose={onRequestClose} contentLabel="Work">
      <button onClick={onRequestClose}>close</button>
      <Title content={title} />
      <Picture imgUrl={thumbnail} />
      <Text>{description}</Text>
      <Title content={detailsTitle} />
      <Text>{details}</Text>
    </Modal>
  );
};
