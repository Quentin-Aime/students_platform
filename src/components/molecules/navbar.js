import React from "react";
import styled from "styled-components";
import { Anchor } from "../atoms/anchor";
import { Logo } from "../atoms/logo";
import { Layout } from "../atoms/layout";

export const Navbar = () => {
  const links = [
    {
      label: "Home",
      url: "home",
    },
    {
      label: "About",
      url: "about",
    },
    {
      label: "Works",
      url: "works",
    },
    {
      label: "Contact",
      url: "contact",
    },
  ];

  return (
    <Layout>
      <Container>
        <Logo />
        <Menu>
          {links.map((link, i) => (
            <Anchor
              label={link.label}
              url={link.url}
              key={i}
              style={{ marginRight: "2rem" }}
            />
          ))}
        </Menu>
      </Container>
    </Layout>
  );
};

const Container = styled.div`
  width: calc(100% - 190px);
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Menu = styled.nav`
  display: flex;
  justify-content: space-between;
`;
