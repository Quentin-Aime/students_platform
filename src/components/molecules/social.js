import React from "react";
import styled from "styled-components";
import { color } from "../../assets/const";

export const Social = ({ logo, label, link }) => {
  return (
    <Container>
      <a href={link}>
        <SocialStyle>
          <div>
            <img src={logo} alt="logo" />
          </div>
          <SocialLabel>{label}</SocialLabel>
        </SocialStyle>
      </a>
    </Container>
  );
};

const Container = styled.div`
  a {
    text-decoration: none;
  }
`;

const SocialStyle = styled.button`
  width: 200px;
  padding: 0.5rem 1rem;
  display: flex;
  align-items: center;
  margin: 1rem;
  border: none;
  background: ${color.secondary};
  font-size: 1rem;
  text-align: center;
`;

const SocialLabel = styled.div`
  color: ${color.primary};
  text-transform: uppercase;
  margin-left: 1rem;
`;
