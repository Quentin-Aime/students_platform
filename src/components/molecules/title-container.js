import React from "react";
import styled from "styled-components";

export const TitleContainer = ({ children }) => (
  <Container>{children}</Container>
);

const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 10rem;
  margin: 3rem 0;
`;
