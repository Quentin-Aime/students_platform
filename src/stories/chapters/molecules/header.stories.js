import React from "react";
import { storiesOf } from "@storybook/react";
import { Header } from "../../../components/molecules/header";

export const doc = `
\`\`\`
import { Header } from "../../../components/molecules/header";
\`\`\`
`;

storiesOf("Molecules|Header", module)
  .addParameters({
    readme: {
      content: doc,
    },
  })
  .add("default", () => {
    return (
      <Header />
    );
  });
