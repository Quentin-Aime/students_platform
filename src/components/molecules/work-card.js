import React, { useState } from "react";
import styled from "styled-components";
import { Title } from "../atoms/title";
import { Picture } from "../atoms/picture";
import { CustomModal } from "./modal";

export const WorkCard = ({
  imgUrl,
  title,
  description,
  detailsTitle,
  details,
}) => {
  const [openModal, setOpenModal] = useState(false);
  return (
    <Container onClick={() => setOpenModal(!openModal)}>
      <Picture imgUrl={imgUrl} />
      <CustomTitle content={title} />
      <CustomModal
        title={title}
        thumbnail={imgUrl}
        isOpen={openModal}
        onRequestClose={() => setOpenModal(false)}
      />
    </Container>
  );
};

const Container = styled.div`
  width: 500px;
  display: flex;
  flex-direction: column;
`;

const CustomTitle = styled(Title)`
  margin-top: 2rem;
`;
