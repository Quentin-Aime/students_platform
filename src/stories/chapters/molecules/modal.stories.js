import React from "react";
import styled from "styled-components";
import { storiesOf } from "@storybook/react";
import { CustomModal } from "../../../components/molecules/modal";
import image from "../../../assets/images/work-1.jpg";

export const doc = `
\`\`\`
import { CustomModal } from "../../../components/molecules/modal";
\`\`\`
`;

const ModalContainer = styled.div`
  display: flex;
  justify-content: center;

  background-color: rgba(0, 0, 0, 0.6);
  padding: 6rem 0;

  & > div {
    width: 50%;
  }
`;

storiesOf("Molecules|Modal", module)
  .addParameters({
    readme: {
      content: doc,
    },
  })
  .add("default", () => {
    return (
      <ModalContainer>
        <CustomModal
          title="Modal title"
          thumbnail={image}
          description="Modal description"
        />
      </ModalContainer>
    );
  });
