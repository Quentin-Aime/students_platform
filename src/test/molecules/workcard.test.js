import React from "react";
import { create } from "react-test-renderer";
import { WorkCard } from "../../components/molecules/work-card";

describe("Workcard component", () => {
  test("Matches the snapshot", () => {
    const card = create(
      <WorkCard title="test" description="Test description" />
    );
    expect(card.toJSON()).toMatchSnapshot();
  });
});
