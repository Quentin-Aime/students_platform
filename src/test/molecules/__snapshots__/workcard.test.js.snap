// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component Matches the snapshot 1`] = `
<div
  className="sc-AxhUy iIDmId"
  onClick={[Function]}
>
  <div
    className="sc-AxiKw IALJF"
  >
    <img
      alt="background"
    />
  </div>
  <h2
    className="sc-AxirZ jOuzpb"
  />
</div>
`;

exports[`Workcard component Matches the snapshot 1`] = `
<div
  className="sc-AxhUy iIDmId"
  onClick={[Function]}
>
  <div
    className="sc-AxiKw IALJF"
  >
    <img
      alt="background"
    />
  </div>
  <h2
    className="sc-AxirZ jOuzpb"
  >
    test
  </h2>
</div>
`;
